sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/ConsoleWriter.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.streaming.sources
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.sources.v2.writer.{DataSourceV2Writer, DataWriterFactory, WriterCommitMessage}
+import org.apache.spark.sql.types.StructType
+
+class ConsoleWriter(batchId: Long, schema: StructType, options: Map[String, String])
+    extends DataSourceV2Writer with Logging {
+  // Number of rows to display, by default 20 rows
+  private val numRowsToShow = options.get("numRows").map(_.toInt).getOrElse(20)
```

Comment:
I am not sure `get("numRows")` on this options map is case-insensitive. Why not just pass on the DataSoureV2Options directly to this writer and use that directly? That is already case-insensitive.

In fact that same pattern should be used for all v2 reader/writer (verify this for Kafka continuous).

Reply:
ConsoleRelation creates this map from a DataSoureV2Options, it contains lowercased keys. 
Using DataSoureV2Options or asking for "numrows" would both work, but with DataSoureV2Options 
`options.get("numRows").map(_.toInt).getOrElse(20)` 
could also be simplified to 
`options.getInt("numRows", 20)`

Reply:
It's not that easy for Kafka continuous, because we're feeding the maps into utility methods (and Kafka-level interfaces) which insist on a java.util.Map[String, Object]. Fortunately the parameters already appear to be case sensitive there, and I think we have tests verifying that various parameters can be specified.